File: sdk-python/tests/iwfcompat/rpc_with_memo_test.py
# Copyright (c) 2026 Super Durable, Inc.
#
# Licensed under the Super Durable Source License 1.0.
# You may not use this file except in compliance with the License.
# See the LICENSE file in the repository root.
#
# SPDX-License-Identifier: LicenseRef-Super-Durable-1.0

from dex import Client

from . import iwf_flows


def compile_memo_replacement(client: Client) -> None:
    flow = iwf_flows.RPC_MEMO_REPLACEMENT
    client.start_flow(flow, "rpc-cache", 0)
    client.invoke_rpc(flow.set_data, "rpc-cache", "value")
    data: str = client.invoke_rpc(flow.get_data, "rpc-cache")
    client.invoke_rpc(flow.set_keyword, "rpc-cache", "keyword")
    keyword: str = client.invoke_rpc(flow.get_keyword, "rpc-cache")
    result: int = client.invoke_rpc(flow.function_one, "rpc-cache", "input")
    del data, keyword, result

File: sdk-typescript/test/iwfcompat/rpc_with_memo_test.ts
// Copyright (c) 2026 Super Durable, Inc.
//
// Licensed under the Super Durable Source License 1.0.
// You may not use this file except in compliance with the License.
// See the LICENSE file in the repository root.
//
// SPDX-License-Identifier: LicenseRef-Super-Durable-1.0

import { type Client } from "../../src/index.js";

import * as flows from "./iwf_flows.js";

export async function compileMemoReplacement(client: Client): Promise<void> {
  const flow = flows.RPC_MEMO_REPLACEMENT;
  await client.startFlow(flow, "rpc-cache", 0);
  await client.invokeRPC(flow.setData, "rpc-cache", "value");
  const data: string = await client.invokeRPC(flow.getData, "rpc-cache");
  await client.invokeRPC(flow.setKeyword, "rpc-cache", "keyword");
  const keyword: string = await client.invokeRPC(flow.getKeyword, "rpc-cache");
  const result: number = await client.invokeRPC(
    flow.functionOne,
    "rpc-cache",
    "input",
  );
  void data;
  void keyword;
  void result;
}
